emily/handler/src/database/accessors.rs

```diff
@@ -696,6 +696,69 @@ pub async fn set_api_state(context: &EmilyContext, api_state: &ApiStateEntry) ->
 
 // Limits ----------------------------------------------------------------------
 
+/// Returns height of oldest stacks block anchored to given bitcoin block
+async fn get_oldest_stacks_block_for_bitcoin_block(
+    context: &EmilyContext,
+    bitcoin_height: u64,
+) -> Result<u64, Error> {
+    query_with_partition_key::<ChainstateByBitcoinHeightTableSecondaryIndex>(
+        context,
+        &bitcoin_height,
+        None,
+        None,
+    )
+    .await?
+    .0
+    .iter()
+    .min_by_key(|entry| entry.key.height)
+    .map(|entry| entry.key.height)
+    .ok_or(Error::NotFound)
+}
+
+async fn calculate_sbtc_left_for_withdrawals(
+    context: &EmilyContext,
+    rolling_withdrawal_blocks: Option<u64>,
+    rolling_withdrawal_cap: Option<u64>,
+) -> Result<u64, Error> {
+    if rolling_withdrawal_blocks.is_none() || rolling_withdrawal_cap.is_none() {
+        return Err(Error::NotFound);
+    }
```

Comment:
Let's essentially unwrap these here by matching on `(Some(_), Some(_))`. That way we don't need to unwrap later. Unwrapping is okay because we have this check, but better to not have to do it at all, since seeing unwrap should set off alarm bells.

Reply:
Also, `Error::NotFound` seems a bit strange as the error variant here, especially since we use the same one for a missing bitcoin chain tip.